dashboard/src/components/LinkGroup/LinkGroup.tsx

```diff
@@ -1,35 +1,58 @@
-import { ReactElement, useMemo } from 'react';
+import { ElementType, Fragment, memo, ReactElement, ReactNode } from 'react';
 
-import LinkWithIcon from '../LinkWithIcon/LinkWithIcon';
+import LinkWithIcon from '@/components/LinkWithIcon/LinkWithIcon';
+import { MessagesKey } from '@/locales/messages';
 
-interface ILinkGroup {
+export interface ILinkGroup {
   links: (ILink | undefined)[];
 }
 
 interface ILink {
   linkText: JSX.Element;
-  title?: ReactElement;
+  title: MessagesKey;
   link?: string;
   icon?: ReactElement;
+  wrapperComponent?: ElementType<{ children: ReactNode }>;
 }
 
+type LinkGroupContainerProps = {
+  children: ReactNode;
+};
+
+export const LinkGroupFilling = ({ links }: ILinkGroup): JSX.Element => {
+  return (
+    <>
+      {links?.map(link => {
+        if (link) {
+          const WrapperComponent = link.wrapperComponent ?? Fragment;
+          return (
+            <WrapperComponent key={link.title}>
+              <LinkWithIcon
+                title={link.title}
+                icon={link.icon}
+                linkText={link.linkText}
+                link={link.link}
+              />
+            </WrapperComponent>
+          );
+        }
+      })}
```

Comment:
what if we move this logic outside the return function? Are the links kind of stable? Otherwise we should also use `useMemo` for the links

Reply:
we are memoizing it here 
`const MemoizedLinkGroupFilling = memo(LinkGroupFilling);`